src/librustc_codegen_ssa/mir/place.rs

```diff
@@ -421,8 +421,15 @@ impl<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 }
             }
             mir::Place::Static(box mir::Static { def_id, ty }) => {
+                // NB: The layout of a static may be unsized as is the case when working
+                // with a static that is an extern_type.
                 let layout = cx.layout_of(self.monomorphize(&ty));
-                PlaceRef::new_sized(bx.get_static(def_id), layout, layout.align.abi)
+                PlaceRef {
+                    llval: bx.get_static(def_id),
+                    llextra: None,
+                    layout: layout,
+                    align: layout.align.abi
+                }
```

Comment:
This does feel a bit reckless to me. Would it be better to instead make the change [here](https://github.com/rust-lang/rust/blob/master/src/librustc_codegen_ssa/mir/rvalue.rs#L380) so that this is only allowed in the case of a r-value that is a reference?

Reply:
I think the right thing to do is to create a new constructor method (`new_thin_place` maybe?) that uses `!type_has_metadata` instead of `is_sized` in the assertion.

Reply:
@oli-obk Added `PlaceRef::new_thin_place`. It is only used in `place.rs` currently, so I didn't make the function `pub`.